### ISSUE TITLE
cmake gets wrong version of opencv

### ISSUE THREAD

**Issue by AbdealiJK** [Itseez/opencv#5169]
I am using arch linux and installed opencv using pacman. It installed version 2.4.11 - I know this because:

 - `ls /usr/lib/libopencv*.so*` gives files with 2.4.11 at the end
 - `pkg-config --modversion opencv` gives 2.4.11

But when I run cmake (I am building something using ros+gazebo) it searches for `/usr/lib/libopencv_videostab.so.2.4.10`
In the `build` directory of my catkin workspace - I found loads of mentions of `2.4.10` and non for `2.4.11`.
I also did a grep inside the source to check if theyve defined `2.4.10` hardcoded somewhere - but I did not find anything in it.

**Comment by AbdealiJK** [comment]
So, I tried downgrading my opencv version, and changed it to opencv-2.4.10-3
And now it searches for `libopencv_videostab.so.2.4.11` :| (so it's definitely not hardcoded)

I did this 2-3 times, so it is reproducable for me

**Comment by alalek** [comment]
Possible problem of this strange behaviour is in CMake cache system: CMake stores somewhere exact version and didn't recheck this again. Try to build from scratch (clean build directory, or remove CMakeCache.txt files if your builds are "in-source").

**Comment by AbdealiJK** [comment]
I dont think that is the problem - because I deleted the build directory both times.

I tried deleting all built directories and just used the src. I also searched for CMakeCache.txt using `find` and got no results.
Still tries to link with the wrong version.

I also checked my OpenCVConfig.cmake - and the version over there is correct. So, I am not sure if this is a opencv issue or a cmake issue or a ROS issue anymore.
Any other ideas on how I could debug this ?
